cpp/src/gandiva/precompiled/extended_math_ops_test.cc

```diff
@@ -273,4 +324,137 @@ TEST(TestExtendedMathOps, TestTrigonometricFunctions) {
+}
+
+TEST(TestExtendedMathOps, TestFloor) {
+  // Ceil functions
+  VerifyFuzzyEquals(floor_float32(0), floor(0));
+  VerifyFuzzyEquals(floor_float64(0), floor(0));
+
+  VerifyFuzzyEquals(floor_float32(-5), floor(-5));
+  VerifyFuzzyEquals(floor_float64(-5), floor(-5));
+
+  VerifyFuzzyEquals(floor_float32(-2371041), floor(-2371041));
+  VerifyFuzzyEquals(floor_float64(-2371041), floor(-2371041));
+
+  VerifyFuzzyEquals(floor_float32(5.45), floor(5.45));
+  VerifyFuzzyEquals(floor_float64(5.45), floor(5.45));
+
+  VerifyFuzzyEquals(floor_float32(-3600.50), floor(-3600.50));
+  VerifyFuzzyEquals(floor_float64(-3600.50), floor(-3600.50));
+}
+
+TEST(TestExtendedMathOps, TestConstants) {
+  // Constants functions
+  VerifyFuzzyEquals(pi(), M_PI);
+  VerifyFuzzyEquals(e(), exp(1.0));
+}
+
+TEST(TestExtendedMathOps, TestSqrt) {
+  // Sqrt functions
+  VerifyFuzzyEquals(sqrt_int32(0), sqrt(0));
+  VerifyFuzzyEquals(sqrt_int64(0), sqrt(0));
+  VerifyFuzzyEquals(sqrt_float32(0), sqrt(0));
+  VerifyFuzzyEquals(sqrt_float64(0), sqrt(0));
+
+  VerifyFuzzyEquals(sqrt_int32(5), sqrt(5));
+  VerifyFuzzyEquals(sqrt_int64(5), sqrt(5));
+  VerifyFuzzyEquals(sqrt_float32(5), sqrt(5));
+  VerifyFuzzyEquals(sqrt_float64(5), sqrt(5));
+
+  VerifyFuzzyEquals(sqrt_int32(2371041), sqrt(2371041));
+  VerifyFuzzyEquals(sqrt_int64(2371041), sqrt(2371041));
+  VerifyFuzzyEquals(sqrt_float32(2371041), sqrt(2371041));
+  VerifyFuzzyEquals(sqrt_float64(2371041), sqrt(2371041));
+
+  VerifyFuzzyEquals(sqrt_int32(3600.50), sqrt(3600));
+  VerifyFuzzyEquals(sqrt_int64(3600.50), sqrt(3600));
+  VerifyFuzzyEquals(sqrt_float32(3600.50), sqrt(3600.50));
+  VerifyFuzzyEquals(sqrt_float64(3600.50), sqrt(3600.50));
+}
+
+TEST(TestExtendedMathOps, TestSign) {
+  // Sqrt functions
+  EXPECT_EQ(sign_int32(0), 0);
+  EXPECT_EQ(sign_int64(0), 0);
+  EXPECT_EQ(sign_float32(0), 0);
+  EXPECT_EQ(sign_float64(0), 0);
+
+  EXPECT_EQ(sign_int32(5), 1);
+  EXPECT_EQ(sign_int64(5), 1);
+  EXPECT_EQ(sign_float32(5), 1);
+  EXPECT_EQ(sign_float64(5), 1);
+
+  EXPECT_EQ(sign_int32(2371041), 1);
+  EXPECT_EQ(sign_int64(2371041), 1);
+  EXPECT_EQ(sign_float32(2371041), 1);
+  EXPECT_EQ(sign_float64(2371041), 1);
+
+  EXPECT_EQ(sign_int32(-3600.50), -1);
+  EXPECT_EQ(sign_int64(-3600.50), -1);
+  EXPECT_EQ(sign_float32(-3600.50), -1);
+  EXPECT_EQ(sign_float64(-3600.50), -1);
+}
+
+TEST(TestExtendedMathOps, TestLshiftRshift) {
+  // Lshift functions
+  EXPECT_EQ(lshift_int32(0, 1), 0 << 1);
+  EXPECT_EQ(lshift_int64(0, 1), 0 << 1);
+
+  EXPECT_EQ(lshift_int32(5, 20), 5 << 20);
+  EXPECT_EQ(lshift_int64(5, 20), 5 << 20);
+
+  EXPECT_EQ(lshift_int32(1, 20), 1 << 20);
+  EXPECT_EQ(lshift_int64(1, 20), 1 << 20);
+
+  // Rshift functions
+  EXPECT_EQ(rshift_int32(0, 1), 0 >> 1);
+  EXPECT_EQ(rshift_int64(0, 1), 0 >> 1);
+
+  EXPECT_EQ(rshift_int32(5, 20), 5 >> 20);
+  EXPECT_EQ(rshift_int64(5, 20), 5 >> 20);
+
+  EXPECT_EQ(rshift_int32(1, 20), 1 >> 20);
+  EXPECT_EQ(rshift_int64(1, 20), 1 >> 20);
+}
```

Comment:
It would be worth adding tests for rshift with a negative value.

Reply:
The test was added.